prepare-netlify-dotenv.mjs

```diff
@@ -0,0 +1,84 @@
+// @ts-check
+
+/*
+ * This script contains workarounds for Netlify environment variables. It transforms
+ * variables available at build time and writes them into a dotenv file.
+ * This file is then read by Next.js both at buildtime and runtime.
+ *
+ * The script is launched via netlify.toml.
+ */
+
+// If you see ESLint error in your editor, check out https://github.com/import-js/eslint-plugin-import/issues/2372
+import fs from "node:fs";
+
+/** @type {Record<string, string>} */
+const envToWrite = {};
+
+/*
+ * Netlify UI does not support context-based environment variables.
+ * For example, we cannot configure XYZ=42 for deploy previews (pull requests)
+ * and XYZ=4242 for production (main branch).
+ *
+ * As a workaround, we read Netlify’s CONTEXT env variable and then
+ * map XYZ_PROD or XYZ_PR into XYZ.
+ */
+
+const suffixByContext = {
+  production: "PROD",
+  "deploy-preview": "PR",
+};
+
+const suffix = suffixByContext[process.env.CONTEXT];
```

Comment:
I ditched prefix and kept only suffix

Related to: https://github.com/dtpstat/website/pull/33#discussion_r790341507